src/vm/gcenv.ee.cpp

```diff
@@ -1214,3 +1214,51 @@ void GCToEEInterface::DiagWalkBGCSurvivors(void* gcContext)
 #endif //GC_PROFILING || FEATURE_EVENT_TRACE
 }
 
+void GCToEEInterface::StompWriteBarrierResize(WriteBarrierResizeArgs args)
+{
+    g_card_table = args.new_card_table;
+    ::StompWriteBarrierResize(args.is_runtime_suspended, args.requires_upper_bounds_check);
+
+    // We need to make sure that other threads executing checked write barriers
+    // will see the g_card_table update before g_lowest/highest_address updates.
+    // Otherwise, the checked write barrier may AV accessing the old card table
+    // with address that it does not cover. Write barriers access card table 
+    // without memory barriers for performance reasons, so we need to flush 
+    // the store buffers here.
+    FlushProcessWriteBuffers();
+
+    g_lowest_address = args.new_lowest_address;
+    VolatileStore(&g_highest_address, args.new_highest_address);
+}
+
+void GCToEEInterface::StompWriteBarrierEphemeral(WriteBarrierEphemeralArgs args)
+{
+    // It is not correct to update g_ephemeral_low and g_ephemeral_high when
+    // using Server GC, since there is no single ephemeral generation across
+    // all of the heaps. Server GC write barriers do not reference these two
+    // globals, but ErectWriteBarrier does.
+    if (!GCHeapUtilities::IsServerHeap())
```

Comment:
I think the IsServerHeap check should to done on the GC side. The VM should just receive the parameters and not apply any policy to it.